Goobi/src/org/goobi/io/SafeFile.java

```diff
@@ -0,0 +1,330 @@
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Linking this library statically or dynamically with other modules is making a
+ * combined work based on this library. Thus, the terms and conditions of the
+ * GNU General Public License cover the whole combination. As a special
+ * exception, the copyright holders of this library give you permission to link
+ * this library with independent modules to produce an executable, regardless of
+ * the license terms of these independent modules, and to copy and distribute
+ * the resulting executable under terms of your choice, provided that you also
+ * meet, for each linked independent module, the terms and conditions of the
+ * license of that module. An independent module is a module which is not
+ * derived from or based on this library. If you modify this library, you may
+ * extend this exception to your version of the library, but you are not obliged
+ * to do so. If you do not wish to do so, delete this exception statement from
+ * your version.
+ */
+package org.goobi.io;
+
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.log4j.Logger;
+
+import de.sub.goobi.helper.Helper;
+
+/**
+ * A {@link java.io.File} delegate class that does not return {@code null} from
+ * its {@code list()} functions. It also provides method calls of
+ * {@link org.apache.commons.io.FileUtils} for an object oriented usage.
+ * 
+ * @author Matthias Ronge
+ */
+public class SafeFile implements Comparable<SafeFile> {
+
+	private static final SafeFile[] NO_FILE = new SafeFile[0];
+
+	private static final String[] NO_STRING = new String[0];
+
+	public static List<SafeFile> createAll(List<File> files) {
+		ArrayList<SafeFile> result = new ArrayList<SafeFile>(files.size());
+		for (File file : files) {
+			result.add(new SafeFile(file));
+		}
+		return result;
+	}
+
+	public static SafeFile createTempFile(String prefix, String suffix) throws IOException {
+		return new SafeFile(File.createTempFile(prefix, suffix));
+	}
+
+	private final File delegate;
+
+    private static final Logger LOGGER = Logger.getLogger(Helper.class);
+
```

Comment:
Indent is wrong and logger use wrong class (Helper.class instead of SafeFile.class).  Statement should / could be moved up to other private statements.